### PR TITLE
chore: use files-from-path module

### PR DESCRIPTION
**Pull request by vasco-santos**
Removes `globSource` code in favour of https://github.com/web3-storage/glob-to-files which is essentially a copy of it for now.

This also aims to make it easier to use the client in Node.js and read files from `fs`. However, given that `client.put` receives an iterable of Files, we can even make this easier by making `glob-to-files` to also export a function with the final files list as an array.

This way we could do in Node.js:

```
await client.put(globToFiles(fsPath))
```

The async iterator is probably nice to have, so I would export both function.

Accepting better names for the module (Friday is even a more difficult day to name things)

Let me know your thoughts @olizilla

### CODE REVIEW

**Comment by alanshaw** [comment]
My vote would be to avoid the word "glob" if possible...I think we kinda want to accept file path(s) and get back file objects, but we support "globs" in those paths. Something like `filepathToFile`/`filesFromPath` (needs work), but the idea is that we make it easy to convert a _file path_ to a _file object(s)_, but the path can be a directory or a glob.

**Comment by vasco-santos** [comment]
`filesFromPath` seems a good naming

**Review by olizilla — APPROVED** [review]
That is neat! My only (very minor) thought is `files-from-path` doesn't feel like it's related to `web3-storage` so might look a little odd in the docs when explaining how to do this for node devs.

```js
import { filesFromPath } from '@web3-stroage/files'
```

might tie it in better or just exposing it as an export from `web3.storage`

```js
import { Web3Storage, filesFromPath } from 'web3.storage'
```

but not a blocker here, just surfacing my feels.

**Comment by olizilla** [comment]
also note that `filesFromPath` does not strictly return "Files" in the web sense, but Filelike things that can have many more properties. Again, I don't think we should use the term `Filelike` here, and this PR is probably fine as is, just surfacing thoughts.

**Comment by vasco-santos** [comment]
I agree @olizilla ! I will create a follow up PR adding it to the client and exporting it for users

**Comment by olizilla** [comment]
It of course might be a good thing that `files-from-path` looks like a totally speperate modules, as it would be fine for node devs to bring any module they like to recurisvely find all the paths in this dir and map them to something that `web3.storage.put` can handle.

@terichadbourne do you have any feels on this? This PR is providing a module to make it easy for a node dev to pass all the files, recursively, in a directory to the `web3.storage` js client `put` method.

**Comment by vasco-santos** [comment]
I think it should be a separate module, but we should also export it from web3storage, so that from the user stand point the objective of it is clear.

So, I would go as you suggest:

```js
import { Web3Storage, filesFromPath } from 'web3.storage'
```

where basically web3.storage imports and exports the util

**Comment by vasco-santos** [comment]
I added it to the client already

File: packages/client/src/platform.web.js
Comment (alanshaw) [inline packages/client/src/platform.web.js:13]: Perhaps these should be more like https://github.com/ipld/js-car/blob/master/lib/indexed-reader-browser.js so that anyone trying to use them in the browser is given a reason why they're not there.

File: packages/client/src/platform.js
Comment (alanshaw) [inline packages/client/src/platform.js:16]: What is the difference?

Comment (vasco-santos) [reply packages/client/src/platform.js:16]: `getFilesFromPath` iterates the `filesFromPath` async iterator and returns an array with all the files. I think users will likely rely on the `getFilesFromPath`, but in the CLI we do a transformation so the iterator is also nice for such use cases.

File: packages/client/src/platform.web.js
Comment (vasco-santos) [reply packages/client/src/platform.web.js:13]: yes, this is a good idea!

**Comment by yusefnapora** [comment]
Nice! I'll add this to the storage how-to once it lands.

**Review by alanshaw — APPROVED** [review]
LGTM - it would be awesome to get a new node.js example that uses this.

**Comment by vasco-santos** [comment]
> LGTM - it would be awesome to get a new node.js example that uses this.

Yes! I am waiting on this to be merged to add it here: https://github.com/web3-storage/web3.storage/pull/103

**Comment by vasco-santos** [comment]
I will rebase this PR now and merge it 🎉

**Pull request merged**
